packages/components/radio-group/src/radio-group.js

```diff
@@ -0,0 +1,143 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import css from './radio-group.css';
+import { getDirectionByKey, customEventNames, directionTypes, getNextElementIndex } from './utils';
+
+import '@tradeshift/elements.radio';
+
+customElementDefineHelper(
+	'ts-radio-group',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				value: { type: String, reflect: true },
+				title: { type: String, reflect: true },
+				index: { type: Number, reflect: true },
+				focused: { type: Boolean, reflect: true },
+				dir: { type: String, reflect: true }
+			};
+		}
+
+		get direction() {
+			return this.dir ? this.dir : this.bodyDir;
+		}
+
+		get radioElements() {
+			return Array.from(this.shadowRoot.host.querySelectorAll('ts-radio')).filter(tab => tab.parentNode === this);
+		}
+
+		toggle(radioElement) {
+			if (!radioElement || radioElement.disabled) {
+				return;
+			}
+
+			for (const radio of this.radios) {
+				radio.checked = radio === radioElement;
+				if (radio.checked) {
+					const index = this.radios.indexOf(radio);
+					const value = radio.value;
+					this.index = index;
+					this.value = value;
+					this.dispatchCustomEvent(customEventNames.RADIO_SELECTED, { radio, index, value });
+				}
+			}
+		}
+
+		firstUpdated() {
+			this.shadowRoot.host.addEventListener(customEventNames.RADIO_CLICK, this.handleRadioClick);
+			this.shadowRoot.host.addEventListener(customEventNames.RADIO_FOCUS, this.handleFocus);
+			this.shadowRoot.host.addEventListener(customEventNames.RADIO_BLUR, this.handleBlur);
+			this.handleSlotChange();
+		}
+
+		disconnectedCallback() {
+			this.shadowRoot.host.removeEventListener(customEventNames.RADIO_CLICK, this.handleRadioClick);
+			this.shadowRoot.host.removeEventListener(customEventNames.RADIO_FOCUS, this.handleFocus);
+			this.shadowRoot.host.removeEventListener(customEventNames.RADIO_BLUR, this.handleBlur);
+		}
+
+		handleRadioClick(e) {
+			e.stopPropagation();
+			this.toggle(e.target);
+		}
+
+		handleFocus(e) {
+			e.stopPropagation();
+			this.focused = true;
+		}
+
+		handleBlur(e) {
+			e.stopPropagation();
+			this.focused = false;
+		}
+
+		handleSlotChange() {
+			this.radios = this.radioElements;
+
+			this.index = 0;
+			for (let i = 0; i < this.radios.length; i++) {
+				const radio = this.radios[i];
+				radio.dir = this.direction;
```

Comment:
If the direction of _ts-radio-group_ changed in runtime, we need to rerender all slotted _ts-radio_ components to apply direction change. Is there any possibility to force rendering of children components without changing their properties?

Reply:
For direction we talked and decided it's okay to not be super responsive to the changes like this since they are really rare 👍 most of the times users are seeing these changes after changing the language and need to re-login to see changes.